### PR TITLE
LOG-3312: Support http output for fluentd

### PR DESCRIPTION
**Pull request by vimalk78**
### Description
 *  Generate fluentd conf for http output
 *  Added unit tests
 * Added functional tests to send logs from fluentd to vector over http
 * Added functional tests to send logs from fluentd to fluentd over http

/cc 
/assign 

/cherry-pick 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-3312

### CODE REVIEW

File: internal/generator/fluentd/output/http/http.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}
+	return KV("headers", utils.ToHeaderStr(o.Http.Headers, "%q:%q"))
+}
+
+func Method(h *logging.Http) string {
+	if h == nil {
+		return "post"
+	}
+	switch h.Method {
+	case "GET":
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply internal/generator/fluentd/output/http/http.go:87]: it is upto the user. mostly it should be `POST` or `PUT`.

File: docs/reference/operator/api.adoc [hunk 2]
```diff
@@ -148,6 +148,7 @@ Output defines a destination for log messages.
 |loki|object|  *(optional)* 
 |googleCloudLogging|object|  *(optional)* 
 |splunk|object|  *(optional)* 
+|http|object|  *(optional)*
```
Comment (jcantrill) [inline docs/reference/operator/api.adoc:151]: is this a manual edit to this file or generated from the make target?

Comment (vimalk78) [reply docs/reference/operator/api.adoc:151]: perhaps it was generated from Makefile

File: internal/generator/fluentd/output/http/userpass.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package http
+
+import (
+	security2 "github.com/openshift/cluster-logging-operator/internal/generator/fluentd/output/security"
+)
+
+type UserNamePass security2.UserNamePass
+
+func (up UserNamePass) Name() string {
+	return "lokiUsernamePasswordTemplate"
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply internal/generator/fluentd/output/http/userpass.go:10]: updated

File: internal/generator/fluentd/output/http/cert-key.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package http
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/generator/fluentd/output/security"
+)
+
+type TLSKeyCert security.TLSCertKey
+
+func (kc TLSKeyCert) Name() string {
+	return "lokiCertKeyTemplate"
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply internal/generator/fluentd/output/http/cert-key.go:10]: updated

File: internal/generator/fluentd/output/http/ca-file.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package http
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/generator/fluentd/output/security"
+)
+
+type CAFile security.CAFile
+
+func (ca CAFile) Name() string {
+	return "lokiCAFileTemplate"
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply internal/generator/fluentd/output/http/ca-file.go:10]: updated

**Pull request merged**